In compatibility doc, fix BrickColor concern and add section on strings with ]]> (#300)
rojo-rbx/rbx-dom: docs/compatibility.md

### BrickColor Properties

> Roblox and `rbx-xml` serialize BrickColor properties using an element named `int`. When using a reflection database, this is fine because those properties get converted to BrickColor during deserialization. However, if the database used by `rbx-xml` is out of date, it won't know to do this, so any new BrickColor properties will be deserialized as integers.

Reviewer: Since Roblox and rbx-dom are the same here, is it necessary to say anything about `int` elements or their conversion to `BrickColor`? Maybe we could just describe the interaction with the reflection database?

Author: After giving this some thought, I agree that mentioning the reflection database interaction should be the main point of this section. I've written a revision.

This is not expected to be a problem, as Roblox will translate the properties just fine when deserializing files, but it may be inconvenient for end users.

### Strings that contain `]]>`

> String properties that contain leading or trailing whitespace characters are serialized using `CDATA` to preserve it. However, the sequence `]]>` is used to end `CDATA` values. If that sequence is present in a string property that also has trailing whitespace, it will serialize incorrectly as a result.

Reviewer: ```suggestion
String properties that contain leading or trailing whitespace characters are serialized using `CDATA` to preserve the whitespace. However, the sequence `]]>` is used to end `CDATA` sections. This means that when a string property contains leading or trailing whitespace, and also contains `]]>`, the property will serialize incorrectly.
```
